franco/Project/org.fundacionjala.automationproject/src/main/java/org/fundacionjala/automationproject/ui/PurchasePage.java

```diff
@@ -0,0 +1,224 @@
+    creditCardType.click();
+    Select day = new Select(creditCardType);
+    day.selectByVisibleText(cardType);
+    creditCardNumber.sendKeys(cardNumber);
+    ccFirstName.sendKeys(fName);
+    ccMiddletName.sendKeys(mName);
+    ccLastName.sendKeys(lName);
+  }
+
+  /**
+   * <p>This method clears billing address form fields.</p>
+   */
+  public void clearBillingAddressFormFields() {
+    billingAddress.clear();
+    billingCity.clear();
+    billingState.clear();
+    billingPostalCode.clear();
+  }
+
+  /**
+   * <p>This method fills the billing address information.</p>
+   *
+   * @param bAddress    is the billing address.
+   * @param bCity       is the billing address city.
+   * @param bState      is the billing address state.
+   * @param bPostalCode is the billing address zip.
+   * @param bCountry    is the billing address country.
+   */
+  public void setBillingAddressInfo(String bAddress, String bCity, String bState, String bPostalCode, String bCountry) {
+    billingAddress.sendKeys(bAddress);
+    billingCity.sendKeys(bCity);
+    billingState.sendKeys(bState);
+    billingPostalCode.sendKeys(bPostalCode);
+    billingCountry.click();
+    Select country = new Select(billingCountry);
+    country.selectByVisibleText(bCountry);
+  }
+
+  /**
+   * <p>This method checks if billing address form is not empty.</p>
+   *
+   * @return whether the form is empty or not.
+   */
+  public boolean billingAddressFormIsFilled() {
+    boolean result = false;
+    if (!billingAddress.getText().equals("")
+      && !billingCity.getText().equals("")
+      && !billingState.getText().equals("")
+      && !billingPostalCode.getText().equals("")) {
+      result = true;
+    }
+    return result;
+  }
+
+  /**
+   * <p>This method clears delivery address form fields.</p>
+   */
+  public void clearDeliveryAddressFormFields() {
+    deliveryAddress.clear();
+    deliveryCity.clear();
+    deliveryState.clear();
+    deliveryPostalCode.clear();
+  }
+
+  /**
+   * <p>This method sets the delivery address information.</p>
+   *
+   * @param dAddress    is the delivery address.
+   * @param dCity       is the delivery address city.
+   * @param dState      is the delivery address state.
+   * @param dPostalCode is the delivery address zip.
+   * @param dCountry    is the delivery address country.
+   */
+  public void setDeliveryAddressInfo(String dAddress, String dCity, String dState, String dPostalCode, String dCountry) {
+    deliveryAddress.sendKeys(dAddress);
+    deliveryCity.sendKeys(dCity);
+    deliveryState.sendKeys(dState);
+    deliveryPostalCode.sendKeys(dPostalCode);
+    deliveryCountry.click();
+    Select country = new Select(deliveryCountry);
+    country.selectByVisibleText(dCountry);
+  }
+
+  /**
+   * <p>This method checks if delivery address form is not empty.</p>
+   *
+   * @return whether the form is empty or not.
+   */
+  public boolean deliveryAddressFormIsFilled() {
```

Comment:
You can user this:
!deliveryAddress.getText().isEmpty();

Reply:
I removed this method.